src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Legend/SemanticTokensLegendEndpoint.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.Semantic.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic.Legend
+{
+    // Used exclusively in VSCode to initialize semantic tokens
+    internal class RazorSemanticTokensLegendEndpoint : IRazorSemanticTokensLegendHandler
+    {
+        private readonly ILogger<RazorSemanticTokensEndpoint> _logger;
+
+        public RazorSemanticTokensLegendEndpoint(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<RazorSemanticTokensEndpoint>();
+        }
+
+        public Task<SemanticTokensLegend> Handle(SemanticTokensLegendParams request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Providing semantic tokens legend.");
```

Comment:
I'm not sure this is adding much value, but I appreciate your efforts 😛

Reply:
😄 you're probably right, although I've found this line helpful in VSCode to verify we're returning a legend since  the line shows up in the Razor Logs.

Reply:
Hmm, nevermind, I could have sworn this showed up in the VSCode-side logs before but I guess not since it's coming from the Razor language server. Since it's not useful, I'll remove

Reply:
I guess I should say, my comment is based on the assumption that we can at least tell from the logs that a request is made, so this line is not adding any extra value. It occurs to me though that without the rich wonderful debugging experience of VS and .NET, maybe that is actually worthwhile, to know that you at least got your request and parameter types right? Or maybe debugging in VS Code is just as good? I have no idea :)